crates/symbolicator/src/test.rs

```diff
@@ -197,6 +160,20 @@ impl Server {
         Self { handle, socket }
     }
 
+    pub fn with_service(service: Service) -> Self {
+        let socket = SocketAddr::from(([127, 0, 0, 1], 0));
+
+        let server =
+            axum::Server::bind(&socket).serve(endpoints::create_app(service).into_make_service());
+
+        let socket = server.local_addr();
+        let handle = tokio::spawn(async {
```

Comment:
Does this need a `CancelOnDrop` wrapper?

(btw, I'd prefer `AbortOnDrop` for that name I think.  matches the tokio naming of aborting a future and is less confusing since we also have `CallOnDrop`)

Reply:
`test::Server` itself does exactly that on `Drop`.